pkg/collections/offledger/storeprovider/olstore.go

```diff
@@ -39,7 +39,7 @@ func newStore(channelID string, dbProvider api.DBProvider, collConfigs map[commo
 		channelID:   channelID,
 		collConfigs: collConfigs,
 		dbProvider:  dbProvider,
-		cache:       cache.New(channelID, dbProvider, config.GetOLCollCacheSize()),
+		cache:       cache.New(channelID, dbProvider, config.GetOLCollCacheSize(), config.GetOLCollCacheEnabled()),
```

Comment:
I don't think we should be creating a cache if it's not enabled. If cache is nil then all retrieval operations should go directly to the DB.